### PR TITLE
Add CodeQL GH Action

### PR DESCRIPTION
**Pull request by hawflau**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hnnasit — APPROVED** [review]
LGTM!

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
```
Comment (mndeveci) [inline .github/workflows/codeql.yml:14]: Sorry if this is a dummy question but just trying to understand our intention here. Why do we need to run this for push to branches as well as on a schedule? Wouldn't be enough if we only have it for pull requests?

Comment (hawflau) [reply .github/workflows/codeql.yml:14]: As stated in the comment in the yml:
```
https://github.com/aws/aws-sam-cli/pull/4552/checks?check_run_id=10533852983
```

I'll keep the on push as well, but limit the branches to only develop and main

**Review by mndeveci — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (mndeveci) [inline .github/workflows/codeql.yml:1]: Can we also configure to run it only for `aws` repository? Like this one here: https://github.com/aws/aws-lambda-builders/blob/develop/.github/workflows/build.yml#L43

**Review by mndeveci — APPROVED** [review]

**Pull request merged**